core/storage/translation/gae_models.py

```diff
@@ -34,6 +35,159 @@
 datastore_services = models.Registry.import_datastore_services()
 
 
+# TODO(#14537): Make sure that domain object files do not import services.
+class TranslatedContentDict(TypedDict):
+    """Dictionary representing TranslatedContent object."""
+
+    content: str|List[str]
+    needs_update: bool
+
+
+class EntityTranslationsModel(base_models.BaseModel):
+    """Model for storing entity translations ."""
+
+    # The id of the corresponding entity.
+    entity_id = datastore_services.StringProperty(required=True, indexed=True)
+    # The type of the corresponding entity.
+    entity_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the corresponding entity.
+    entity_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+    # The ISO 639-1 code for the language an entity is written in.
+    language_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The translated content.
+    translations = datastore_services.JsonProperty(required=True)
+
+    @staticmethod
+    def get_deletion_policy() -> base_models.DELETION_POLICY:
+        """Model doesn't contain any data directly corresponding to a user."""
+        return base_models.DELETION_POLICY.NOT_APPLICABLE
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model does not contain user data."""
```

Comment:
@DubeySandeep I have a question: 
1. How to know whether a data is linked to an user or not ( I've written these docstring by taking guidance from already written code but wants to know in general) ?

Thank you.

Reply:
This is usually through reasoning about what's going on. In this case, there's no user data in any of the fields here since they all relate to explorations and not to any individual user.

Reply:
Thank you, this helps.